docs/roadmap.md

```diff
@@ -0,0 +1,98 @@
+| entity field local file         | with new-mode, allow for field overrides from file         |         | unimplemented |
+| fetch URLs from routes/ingress  | when backstage,third party running on K8s, find URL        |         | unimplemented |
+| flags for output                | allow for output summary vs. json vs. yaml etc.            |         | unimplemented |
+| flags for output files          | new-model store Components/Resources/API in separate files |         | unimplemented |
+| flags for field overrides       | new-model provide field values via command line flags      |         | unimplemented |
+| release process                 | initially github action/goreleaser; eventually konflux     |         | unimplemented |
+
+## Upstream Backstage
+
+### Direct injection of YAML when Creating Entities in the Catalog
+
+While the input format the body supplied to this REST API has a type field, best as we can tell, the only types supported
+are a HTTP accessible URL or a local file.
+
+Will users of the CLI be happy with having to take the extra step of pushing the YAML from `bac new-model ...` to say a
+file hosted in Git repository and then provide that URL to `bac import-model ...` ?
+
+Or are will they be enamored with the idea (albeit it punts on Gitops) with a flow like
+
+```shell
+bac new-model kserve | bac import-model -f -
+```
+
+or 
+
+```shell
+bac new-model kubeflow > catalog-info.yaml
+bac import-model catalog-info.yaml 
+```
+
+#### One advantage of being in Golang with respect to Gitops with GitHub ...
+
+The uber functional `gh` command, [GitHub's official command line tool](https://github.com/cli/cli), happens to be written
+in Golang.
+
+The `gh` command has a A LOT of usability features outside of the standard `git` command.
+
+The `bac new-model ...` invocation could take in the necessary Git credentials and connection parameters and via use of `gh` code as 
+vendored dependencies:
+- create the repository if needed
+- create not just a commit, but a pull request, with the `catalog-info.yaml` file(s)
+- rebase if necessary
+- merge that pull request
+- and then post to the Backstage Catalog REST API to import ... i.e. what we do with `bac import-model ...`
+
+
+## New 'Model Metadata' sources
+
+| Source      | Summary/REST/CRDs                | Questions/Comments                                            | Priority | Tracker | Status  |
+|-------------|----------------------------------|---------------------------------------------------------------|----------|---------|---------|
+| Kubeflow    | Endpoint URL.  Has both REST/CRD | RHOAI Jira marked done.  Which version?  End to end examples? | high     |         | waiting |
+| 3Scale      | All data ready.  Yes REST/CRDs   | Perhaps the next highest item. Devex vs. RHOAI priorities     | high     |         | new     |
+| HuggingFace | All data ready.  REST only       | Direct competitor or co-opetition.  Best for tech docs        |          |         | new     |
```

Comment:
I'd replace the competitor comment with something like "Most popular source for public models."
It doesn't really matter if they're a competitor.